app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,16 @@
+class CommentsController < ApplicationController
+  def create
+    @current_user = current_user
+    @comment = Comment.new(comment_params)
+    @comment.author = current_user
+    @comment.post_id = params[:post_id].to_i
+    @comment.save
+    redirect_to user_post_path(params[:user_id], params[:post_id])
+  end
+
```

Comment:
- [ ] I am unable to create a comment. Currently giving this issue in the terminal while creating a comment.
![Screenshot from 2022-08-21 11-23-43](https://user-images.githubusercontent.com/26767383/185777713-7d80d8bc-c019-4887-bfdf-cc552aecd6f9.png)

- [ ] Also, use the condition while saving the comment just like you did with the post so that the user will know that there is some issue while creating a comment.

Reference:

```ruby
 if @post.save
      redirect_to user_post_path(current_user, @post)
    else
      render :new, status: :unprocessable_entity
    end
    ```